File: deal-module/src/main/java/org/example/dealmodule/deal/domain/Deal.java
package org.example.dealmodule.deal.domain;

import lombok.Builder;
import lombok.With;
import org.example.activitymodule.ActivityInternalDto;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Builder
public record Deal(
        UUID id,
        String title,
        @With DealStage stage,
        String expectedRevenue,
        LocalDateTime expectedCloseDate,
        UUID contact,
        UUID owner,
        @With List<ActivityInternalDto> activities
) {
}

File: ux/src/app/models/deal/deal.ts
import {ActivitySummary} from '../activity/activity-summary';

export type Deal = {
  id?: string,
  title: string,
  stage?: number,
  expectedRevenue?: string,
  expectedCloseDate?: string,
  contact?: string,
  owner: string,
  activities: ActivitySummary[]
}
